### DISCUSSION TITLE
Form customization options

### DISCUSSION THREAD

**Discussion by InitiativeWorking** [burkybang/SuiteAdvanced-Feedback#96] [Ideas]
Hi there!

How possible would it be to add more options into the Custom Form builder?

![image](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/87781169/309da4a7-3c7d-463a-b221-bcfc532ed49e)

Stuff like,
- linking to each custom fields itself
- displaying the custom field ID as column
- adding custom translations to the fields
- seeing/editing the field's help section
- adding some way of seeing what(mass updates, scripts, workflows) is executing over this field

**Comment by burkybang** [comment]
Because the sublist rows can be reordered, it would be a bit tricky to ensure any links or field IDs stay accurate.  At first look, I don't see a good way to do these, but I'll keep thinking about them.
- _linking to each custom fields itself_
- _displaying the custom field ID as column_

I don't see the below three being possible or practical at all.
- _adding custom translations to the fields_
  - There's not really an API that I'm aware of here.
- _seeing/editing the field's help section_
  - Maybe seeing the help section is possible if I can figure out the best way to keep them accurate due to the sublist reorderability, but I don't know about editing it. It also might get pretty cramped if all the help texts were squeezed in there.
- _adding some way of seeing what(mass updates, scripts, workflows) is executing over this field_
  - I agree this would be extremely handy, but I don't think it's very feasible. I suppose it could be possible to search through every unhidden script file for the field ID, but that could be pretty time consuming. I don't see that being possible for mass updates or workflows.

**Comment by burkybang** [comment]
@InitiativeWorking I just submitted a new release to Google with one of your ideas.

> displaying the custom field ID

I've included this as part of the "Show IDs" (formerly "Show Field IDs") option in the SuiteAdvanced menu. It will not only show field IDs but subtab IDs and sublist IDs too on the Customize Form page.

![image](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/927886/ad152358-a19b-4ca4-a318-e1967476f607)

**Comment by InitiativeWorking** [comment]
Hey man! That's awesome, thank you very much 🔥

**Comment by InitiativeWorking** [comment]
Hi @burkybang !

Do you think it would be possible to make the ID copy-able? I'm having some trouble selecting it so I always go to the source code and copy from there

**Comment by burkybang** [comment]
I've been thinking about that too 🤔

A copy button could be a good idea. I'll play around with some ideas, but I agree it's not ideal at the moment.

**Comment by burkybang** [comment]
@InitiativeWorking I figured out a way to make them selectable for copying. It'll be out in the next release.

**Reply by burkybang** [reply]
@InitiativeWorking I just submitted the next release to Google

**Reply by burkybang** [reply]
@InitiativeWorking The new release is now out!

**Comment by InitiativeWorking** [comment]
Let's go! 🔥🔥🔥 great work!
